pymc/distributions/multivariate.py

```diff
@@ -543,16 +543,21 @@ class Multinomial(Discrete):
 
     @classmethod
     def dist(cls, n, p, *args, **kwargs):
-        if isinstance(p, np.ndarray) or isinstance(p, list):
-            if (np.asarray(p) < 0).any():
-                raise ValueError(f"Negative `p` parameters are not valid, got: {p}")
-            p_sum = np.sum([p], axis=-1)
-            if not np.all(np.isclose(p_sum, 1.0)):
+        p = at.as_tensor_variable(p)
+        if isinstance(p, TensorConstant):
+            p_ = np.asarray(p.data)
+            if np.any(p_ < 0):
+                raise ValueError(f"Negative `p` parameters are not valid, got: {p_}")
+            p_sum_ = np.sum([p_], axis=-1)
+            if not np.all(np.isclose(p_sum_, 1.0)):
                 warnings.warn(
-                    f"`p` parameters sum up to {p_sum}, instead of 1.0. They will be automatically rescaled. You can rescale them directly to get rid of this warning.",
+                    f"`p` parameters sum to {p_sum_}, instead of 1.0. They will be "
+                    f"automatically rescaled. You can rescale them directly to get "
+                    f"rid of this warning.",
```

Comment:
```suggestion
                    f"`p` parameters sum to {p_sum_}, instead of 1.0. "
                    "They will be automatically rescaled. "
                    "You can rescale them directly to get rid of this warning.",
```